### PR TITLE
Separate sign up route into host and coordinator sign up routes

### PR DESCRIPTION
**Pull request by erikguntner**
This PR is a part of the ongoing refactoring of the sign up experience and includes:

- Creating api spec and separate routes for host and coordinator at `signup/host` and `signup/coordinator`
- Removing the original sign up route
- Updating the current sign up page to make the request to `signup/host` to keep the app functional

### CODE REVIEW

File: api/openapi_server/controllers/auth_controller.py [hunk 1]
```diff
@@ -123,6 +123,44 @@ def signup():  # noqa: E501
...
+def signUpCoordinator():  # noqa: E501
+    """Signup a new Coordinator
+    """
+    if connexion.request.is_json:
+        body = connexion.request.get_json()
+
+    secret_hash = get_secret_hash(body['email'])
+
+    # Signup user
+    with Session(db_engine) as session:
```
Comment (ju1es) [inline api/openapi_server/controllers/auth_controller.py:135]: why do we store a coordinator and not a host in postgres?

File: api/openapi_server/controllers/auth_controller.py [hunk 2]
```diff
@@ -123,6 +123,44 @@ def signup():  # noqa: E501
...
+        user = db.User(email=body['email'])
+        session.add(user)
```
Comment (ju1es) [inline api/openapi_server/controllers/auth_controller.py:137]: why do we store a coordinator and not a host's email in postgres?

**Comment by ju1es** [comment]
ah is this in preparation for the coordinator invite issue? curious about what the refactoring part is.

File: api/openapi_server/controllers/auth_controller.py (hunk 2 above)
Comment (erikguntner) [reply api/openapi_server/controllers/auth_controller.py:137]: We save the host as well. Not sure why Github decided not to highlight it in the changes but you can see it if you view the whole file

https://github.com/hackforla/HomeUniteUs/blob/2d1d17d37764cc0ecc231e01b4f2e531252853a3/api/openapi_server/controllers/auth_controller.py

**Comment by erikguntner** [comment]
@ju1es The refactoring relates to these two issues:

https://github.com/hackforla/HomeUniteUs/issues/404
https://github.com/hackforla/HomeUniteUs/issues/403

I think two meetings ago we had a discussion and decided it was best to have separate sign up pages and endpoints after the account selection screen instead of using the same form and endpoint for host and coordinator. This is just the backend portion of that

File: api/openapi_server/_spec/openapi.yaml [hunk 3]
```diff
@@ -420,30 +453,21 @@ paths:
...
           application/json:
             schema:
               type: object
               properties:
-                email:
-                  type: string
-                password:
-                  type: string
-                session:
+                username:
```
Comment (mira-kine) [inline api/openapi_server/_spec/openapi.yaml:467]: Are we having them create a username only in sign up, or require it for sign in as well?

**Review by mira-kine — APPROVED** [review]
Looks good, thanks Erik! I just added a comment about what we might be using "username" in the future about sign up/sign in flow, but other than that looks good. I will work on the front end portion in the upcoming week!

File: api/openapi_server/_spec/openapi.yaml (hunk 3 above)
Comment (erikguntner) [reply api/openapi_server/_spec/openapi.yaml]: @mira-kine thanks for pointing this out. It should be email instead of username. I'll switch it

**Pull request merged**